src/Servers/Kestrel/Core/test/AddressBinderTests.cs

```diff
@@ -149,5 +153,39 @@ public async Task FallbackToIPv4WhenIPv6AnyBindFails(string address)
             Assert.True(ipV6Attempt, "Should have attempted to bind to IPAddress.IPv6Any");
             Assert.Contains(logger.Messages, f => f.Equals(CoreStrings.FormatFallbackToIPv4Any(80)));
         }
+
+        [Fact]
+        public async Task DefaultAddressBinderWithoutDevCertButHttpsConfiguredBindsToHttpsPorts()
+        {
+            var x509Certificate2 = TestResources.GetTestCertificate();
+            var logger = new MockLogger();
+            var addresses = new ServerAddressesFeature();
+            var services = new ServiceCollection();
+            services.AddLogging();
+            var options = new KestrelServerOptions()
+            {
+                // This stops the dev cert from being loaded
+                IsDevCertLoaded = true,
+                ApplicationServices = services.BuildServiceProvider()
+            };
+
+            options.ConfigureEndpointDefaults(e =>
+            {
+                if (e.IPEndPoint.Port == 5001)
```

Comment:
If you didn't have `IsDevCertLoaded = true,` above, this test would pass even without the custom ConfigureEndpointDefaults, right?

I think this test would be a little more convincing if it made port 5000 the TLS endpoint.

Reply:
The default binding only checks 5001 for IsTls